Botnet-Honeypot/Honeypot
# Frontend backend integration

- Frontend now asks backend for a target system after attacker logs in.
- When frontend closes proxy connection (for any reason), target system is yielded back to backend.
- Backend address is configurable through `BACKEND_ADDRESS` env var in frontend.
- The returned target system address from backend is configurable through env var `TARGET_SYSTEM_ADDRESS`.
- Backend now waits until target container is fully ready before responding to an acquisition request. This decreases delays caused by the frontend failing and retrying connections to target container.
- If the frontend, for some reason, fails to connect to target container, it will retry the connection with a delay inbetween.

Some issues:
- Connecting with a blank password currently does not work. It seems like Paramiko does not support it very well and the target container is not configured to accept blank passwords.
- Hostname `host.docker.internal` in `dev/docker-compose.yml` does not work on Linux but on Docker Desktop.

How to test:
- Run `docker-compose up` in `dev` folder
- Connect using SSH to frontend `ssh localhost`
- Make sure you can connect, target container starts and you can run commands etc.
